openscap_report/report_generators/html_templates/js/oval_graph_generation_script.js

```diff
@@ -309,46 +415,22 @@ function get_operator_node() {
     node_content.className = "pf-c-tree-view__node-content";
     node_container.appendChild(node_content);
 
-    const node_title = SPAN.cloneNode();
-    node_title.className = "pf-c-tree-view__node-text";
-    node_content.appendChild(node_title);
+    const node_text = SPAN.cloneNode();
+    node_text.className = "pf-c-tree-view__node-text";
+    node_content.appendChild(node_text);
 
-    return { operator_node, node_title };
+    return { operator_node, node_text };
 }
 
 function get_OVAL_tree_operator_node(node_data) {
-    const { operator_node, node_title } = get_operator_node();
-    let color = '';
-    let icon = 'fa-question-circle';
-    if (node_data.value == 'true') {
-        color = 'pf-m-green';
-        icon = 'fa-check';
-    } else if (node_data.value == 'false') {
-        color = 'pf-m-red';
-        icon = 'fa-times';
-    }
-
-    let negate_color = '';
-    let negate_icon = icon;
-    if (node_data.negation) {
-        negate_color = COLOR_TRANSLATION[NEGATION_COLOR[color]];
-        negate_icon = NEGATION_ICON[icon];
-    } else {
-        negate_color = COLOR_TRANSLATION[color];
-    }
-    const node = get_node(negate_color);
-    node_title.appendChild(node);
-    const html_icon = get_icon_as_html(negate_icon);
-    node.appendChild(html_icon);
-    if (node_data.negation) {
-        node.appendChild(get_bold_text("NOT"));
-        html_icon.classList.add("icon-space");
-    }
+    const { operator_node, node_text } = get_operator_node();
+    const { node, color, icon } = base_operator_node(node_data, node_text);
 
     node.appendChild(get_bold_text(` ${node_data.node_type} `));
-    node_title.appendChild(get_label(color, node_data.tag));
-    node_title.appendChild(get_label(color, node_data.value, get_icon_as_html(icon)));
-    node_title.appendChild(get_note(`\u00A0\u00A0${node_data.comment}`));
+    node_text.appendChild(get_label(color, node_data.tag));
+    node_text.appendChild(get_label(color, node_data.value, get_icon_as_html(icon)));
+    node_text.appendChild(get_note(`\u00A0\u00A0${node_data.comment ? node_data.comment : ""}`));
```

Comment:
Because JavaScript interprets `&nbsp;`  as a sequence of characters which are escaped for DOM elements. [Here is an article about that](https://stackoverflow.com/questions/66638682/nbsp-not-displaying-properly-in-javascript). There is an option to insert this string as pure HTML into this DOM element, but I'd like to keep an object-oriented approach to the DOM elements.